# HG changeset patch
# User Rik <rik@octave.org>
# Date 1493404719 25200
#      Fri Apr 28 11:38:39 2017 -0700
# Node ID 73ff72d3d603da0e1319eab48e4490bfbd692060
# Parent  30b6eccd67085dcf0ccdb83361b7bf129cedcd20
maint: Eliminate <cfloat.h> header from libinterp files

* __contourc__.cc, __qp__.cc, data.cc, graphics.cc, load-save.cc, ls-hdf5.cc,
ls-mat-ascii.cc, ls-mat4.cc, ls-mat5.cc, ls-oct-binary.cc, ls-oct-text.h,
mappers.cc, mex.cc, pr-output.cc, sysdep.cc, variables.h, xnorm.cc, __glpk__.cc:
Remove "#include <cfloat>".  Add "#include <limits>" if necessary.  Remove
redundant #includes.  Alphabetize #includes.

diff --git a/libinterp/corefcn/__contourc__.cc b/libinterp/corefcn/__contourc__.cc
--- a/libinterp/corefcn/__contourc__.cc
+++ b/libinterp/corefcn/__contourc__.cc
@@ -31,23 +31,20 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
-
-#include "quit.h"
+#include <limits>
 
 #include "defun.h"
-#include "error.h"
-#include "ovl.h"
+#include "ov.h"
 
 // FIXME: this looks like trouble...
 static Matrix this_contour;
 static Matrix contourc;
 static int elem;
 
 // This is the quanta in which we increase this_contour.
 #define CONTOUR_QUANT 50
diff --git a/libinterp/corefcn/__qp__.cc b/libinterp/corefcn/__qp__.cc
--- a/libinterp/corefcn/__qp__.cc
+++ b/libinterp/corefcn/__qp__.cc
@@ -19,17 +19,19 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
+#include <cmath>
+
+#include <limits>
 
 #include "chol.h"
 #include "svd.h"
 #include "mx-m-dm.h"
 #include "EIG.h"
 
 #include "defun.h"
 #include "error.h"
diff --git a/libinterp/corefcn/data.cc b/libinterp/corefcn/data.cc
--- a/libinterp/corefcn/data.cc
+++ b/libinterp/corefcn/data.cc
@@ -22,31 +22,31 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <sys/types.h>
-
-#include <cfloat>
+#include <cstddef>
+#include <cstdint>
 #include <ctime>
 
+#include <algorithm>
+#include <limits>
 #include <string>
 
 #include "lo-ieee.h"
 #include "lo-math.h"
+#include "mx-base.h"
 #include "oct-base64.h"
+#include "oct-binmap.h"
 #include "oct-time.h"
-#include "str-vec.h"
 #include "quit.h"
-#include "mx-base.h"
-#include "oct-binmap.h"
 
 #include "Cell.h"
 #include "defun.h"
 #include "error.h"
 #include "errwarn.h"
 #include "oct-map.h"
 #include "ovl.h"
 #include "ov.h"
diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -20,20 +20,23 @@ along with Octave; see the file COPYING.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <cctype>
-#include <cfloat>
+#include <cmath>
+#include <cstdint>
 #include <cstdlib>
 
 #include <algorithm>
+#include <iostream>
+#include <limits>
 #include <list>
 #include <map>
 #include <set>
 #include <string>
 #include <sstream>
 
 #include "cmd-edit.h"
 #include "file-ops.h"
diff --git a/libinterp/corefcn/load-save.cc b/libinterp/corefcn/load-save.cc
--- a/libinterp/corefcn/load-save.cc
+++ b/libinterp/corefcn/load-save.cc
@@ -23,27 +23,27 @@ along with Octave; see the file COPYING.
 // Author: John W. Eaton.
 // HDF5 support by Steven G. Johnson <stevenj@alum.mit.edu>
 // Matlab v5 support by James R. Van Zandt <jrv@vanzandt.mv.com>
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
 #include <cstring>
-#include <cctype>
 
 #include <fstream>
 #include <iomanip>
 #include <iostream>
+#include <list>
 #include <sstream>
 #include <string>
 
 #include "byte-swap.h"
+#include "dMatrix.h"
 #include "data-conv.h"
 #include "file-ops.h"
 #include "file-stat.h"
 #include "glob-match.h"
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "oct-env.h"
 #include "oct-locbuf.h"
@@ -65,17 +65,16 @@ along with Octave; see the file COPYING.
 #include "pager.h"
 #include "pt-exp.h"
 #include "symtab.h"
 #include "sysdep.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 #include "version.h"
-#include "dMatrix.h"
 
 #include "ls-hdf5.h"
 #include "ls-mat-ascii.h"
 #include "ls-mat4.h"
 #include "ls-mat5.h"
 #include "ls-oct-text.h"
 #include "ls-oct-binary.h"
 
diff --git a/libinterp/corefcn/ls-hdf5.cc b/libinterp/corefcn/ls-hdf5.cc
--- a/libinterp/corefcn/ls-hdf5.cc
+++ b/libinterp/corefcn/ls-hdf5.cc
@@ -23,21 +23,18 @@ along with Octave; see the file COPYING.
 // Author: Steven G. Johnson <stevenj@alum.mit.edu>
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #if defined (HAVE_HDF5)
 
-#include <cfloat>
-#include <cstring>
 #include <cctype>
 
-#include <fstream>
 #include <iomanip>
 #include <iostream>
 #include <limits>
 #include <string>
 #include <vector>
 
 #include "byte-swap.h"
 #include "data-conv.h"
diff --git a/libinterp/corefcn/ls-mat-ascii.cc b/libinterp/corefcn/ls-mat-ascii.cc
--- a/libinterp/corefcn/ls-mat-ascii.cc
+++ b/libinterp/corefcn/ls-mat-ascii.cc
@@ -19,57 +19,51 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
-#include <cstring>
 #include <cctype>
 
-#include <fstream>
 #include <iomanip>
 #include <iostream>
 #include <sstream>
 #include <string>
 
 #include "byte-swap.h"
+#include "dMatrix.h"
 #include "data-conv.h"
 #include "file-ops.h"
 #include "glob-match.h"
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "oct-env.h"
 #include "oct-time.h"
 #include "quit.h"
-#include "str-vec.h"
 
 #include "Cell.h"
 #include "defun.h"
 #include "error.h"
-#include "errwarn.h"
 #include "interpreter.h"
 #include "lex.h"
 #include "load-save.h"
 #include "ls-ascii-helper.h"
 #include "ls-mat-ascii.h"
-#include "ovl.h"
 #include "oct-map.h"
 #include "ov-cell.h"
+#include "ov.h"
 #include "pager.h"
 #include "pt-exp.h"
 #include "sysdep.h"
-#include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 #include "version.h"
-#include "dMatrix.h"
 
 static std::string
 get_mat_data_input_line (std::istream& is)
 {
   std::string retval;
 
   bool have_data = false;
 
diff --git a/libinterp/corefcn/ls-mat4.cc b/libinterp/corefcn/ls-mat4.cc
--- a/libinterp/corefcn/ls-mat4.cc
+++ b/libinterp/corefcn/ls-mat4.cc
@@ -19,57 +19,50 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
-#include <cstring>
-#include <cctype>
-
-#include <fstream>
 #include <iomanip>
 #include <iostream>
 #include <string>
-#include <vector>
 
 #include "byte-swap.h"
+#include "dMatrix.h"
+#include "dSparse.h"
 #include "data-conv.h"
 #include "file-ops.h"
 #include "glob-match.h"
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "oct-env.h"
+#include "oct-locbuf.h"
 #include "oct-time.h"
 #include "quit.h"
-#include "str-vec.h"
-#include "oct-locbuf.h"
 
+#include "ls-mat4.h"
 #include "Cell.h"
 #include "defun.h"
 #include "error.h"
 #include "errwarn.h"
 #include "load-save.h"
-#include "ovl.h"
 #include "oct-map.h"
 #include "ov-cell.h"
+#include "ovl.h"
 #include "pager.h"
 #include "pt-exp.h"
 #include "sysdep.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 #include "version.h"
-#include "dMatrix.h"
-#include "dSparse.h"
 
-#include "ls-mat4.h"
 
 // Read LEN elements of data from IS in the format specified by
 // PRECISION, placing the result in DATA.  If SWAP is TRUE, swap
 // the bytes of each element before copying to DATA.  FLT_FMT
 // specifies the format of the data if we are reading floating point
 // numbers.
 
 static void
@@ -134,19 +127,18 @@ read_mat_file_header (std::istream& is, 
     return -1;
 
   if (! is.read (reinterpret_cast<char *> (&len), 4))
     return -1;
 
 // If mopt is nonzero and the byte order is swapped, mopt will be
 // bigger than we expect, so we swap bytes.
 //
-// If mopt is zero, it means the file was written on a little endian
-// machine, and we only need to swap if we are running on a big endian
-// machine.
+// If mopt is zero, it means the file was written on a little endian machine,
+// and we only need to swap if we are running on a big endian machine.
 //
 // Gag me.
 
   if (octave::mach_info::words_big_endian () && mopt == 0)
     swap = true;
 
   // mopt is signed, therefore byte swap may result in negative value.
 
diff --git a/libinterp/corefcn/ls-mat5.cc b/libinterp/corefcn/ls-mat5.cc
--- a/libinterp/corefcn/ls-mat5.cc
+++ b/libinterp/corefcn/ls-mat5.cc
@@ -21,68 +21,65 @@ along with Octave; see the file COPYING.
 */
 
 // Author: James R. Van Zandt <jrv@vanzandt.mv.com>
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
 #include <cstring>
-#include <cctype>
-
-#include <fstream>
+
 #include <iomanip>
 #include <iostream>
+#include <limits>
 #include <sstream>
 #include <string>
 #include <vector>
 
 #include "byte-swap.h"
+#include "dMatrix.h"
 #include "data-conv.h"
 #include "file-ops.h"
+#include "file-stat.h"
 #include "glob-match.h"
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "oct-env.h"
+#include "oct-locbuf.h"
 #include "oct-time.h"
 #include "quit.h"
 #include "str-vec.h"
-#include "file-stat.h"
-#include "oct-locbuf.h"
 
 #include "Cell.h"
 #include "call-stack.h"
+#include "defaults.h"
 #include "defun.h"
 #include "error.h"
 #include "errwarn.h"
-#include "load-save.h"
+#include "interpreter-private.h"
+#include "interpreter.h"
 #include "load-path.h"
-#include "ovl.h"
+#include "load-save.h"
+#include "ls-mat5.h"
+#include "ls-utils.h"
 #include "oct-map.h"
 #include "ov-cell.h"
 #include "ov-class.h"
 #include "ov-fcn-inline.h"
+#include "ov.h"
+#include "ovl.h"
 #include "pager.h"
+#include "parse.h"
 #include "pt-exp.h"
 #include "sysdep.h"
-#include "interpreter-private.h"
-#include "interpreter.h"
 #include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 #include "version.h"
-#include "dMatrix.h"
-
-#include "ls-utils.h"
-#include "ls-mat5.h"
-
-#include "parse.h"
-#include "defaults.h"
 
 #if defined (HAVE_ZLIB)
 #  include <zlib.h>
 #endif
 
 #define READ_PAD(is_small_data_element, l) ((is_small_data_element) ? 4 : (((l)+7)/8)*8)
 #define PAD(l) (((l) > 0 && (l) <= 4) ? 4 : (((l)+7)/8)*8)
 #define INT8(l) ((l) == miINT8 || (l) == miUINT8 || (l) == miUTF8)
diff --git a/libinterp/corefcn/ls-oct-binary.cc b/libinterp/corefcn/ls-oct-binary.cc
--- a/libinterp/corefcn/ls-oct-binary.cc
+++ b/libinterp/corefcn/ls-oct-binary.cc
@@ -19,57 +19,43 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
-#include <cstring>
-#include <cctype>
-
-#include <fstream>
-#include <iomanip>
 #include <iostream>
 #include <string>
-#include <vector>
 
 #include "byte-swap.h"
 #include "data-conv.h"
 #include "file-ops.h"
 #include "glob-match.h"
 #include "lo-mappers.h"
 #include "mach-info.h"
 #include "oct-env.h"
+#include "oct-locbuf.h"
 #include "oct-time.h"
-#include "quit.h"
-#include "str-vec.h"
-#include "oct-locbuf.h"
 
-#include "Cell.h"
 #include "defun.h"
 #include "error.h"
 #include "errwarn.h"
 #include "load-save.h"
-#include "ovl.h"
-#include "oct-map.h"
+#include "ls-oct-binary.h"
+#include "ls-utils.h"
 #include "ov-cell.h"
+#include "ov.h"
 #include "pager.h"
 #include "pt-exp.h"
 #include "sysdep.h"
-#include "unwind-prot.h"
 #include "utils.h"
 #include "variables.h"
 #include "version.h"
-#include "dMatrix.h"
-
-#include "ls-utils.h"
-#include "ls-oct-binary.h"
 
 // Extract one value (scalar, matrix, string, etc.) from stream IS and
 // place it in TC, returning the name of the variable.  If the value
 // is tagged as global in the file, return TRUE in GLOBAL.  If SWAP
 // is TRUE, swap bytes after reading.
 //
 // The data is expected to be in the following format:
 //
diff --git a/libinterp/corefcn/ls-oct-text.h b/libinterp/corefcn/ls-oct-text.h
--- a/libinterp/corefcn/ls-oct-text.h
+++ b/libinterp/corefcn/ls-oct-text.h
@@ -20,18 +20,17 @@ along with Octave; see the file COPYING.
 
 */
 
 #if ! defined (octave_ls_oct_text_h)
 #define octave_ls_oct_text_h 1
 
 #include "octave-config.h"
 
-#include <cfloat>
-
+#include <limits>
 #include <sstream>
 #include <string>
 
 #include "str-vec.h"
 
 #include "ls-ascii-helper.h"
 
 class octave_value;
diff --git a/libinterp/corefcn/mappers.cc b/libinterp/corefcn/mappers.cc
--- a/libinterp/corefcn/mappers.cc
+++ b/libinterp/corefcn/mappers.cc
@@ -21,17 +21,16 @@ along with Octave; see the file COPYING.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
 #include <cctype>
-#include <cfloat>
 
 #include "lo-ieee.h"
 #include "lo-specfun.h"
 #include "lo-mappers.h"
 
 #include "defun.h"
 #include "error.h"
 #include "variables.h"
diff --git a/libinterp/corefcn/mex.cc b/libinterp/corefcn/mex.cc
--- a/libinterp/corefcn/mex.cc
+++ b/libinterp/corefcn/mex.cc
@@ -19,28 +19,30 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
-#include <csetjmp>
 #include <cstdarg>
 #include <cstdlib>
 #include <cstring>
 #include <cctype>
 
+#include <limits>
+#include <map>
 #include <set>
+#include <string>
 
 #include "f77-fcn.h"
 #include "lo-ieee.h"
 #include "oct-locbuf.h"
+#include "quit.h"
 
 #include "Cell.h"
 #include "call-stack.h"
 #include "error.h"
 #include "interpreter.h"
 // mxArray must be declared as a class before including mexproto.h.
 #include "mxarray.h"
 #include "mexproto.h"
diff --git a/libinterp/corefcn/pr-output.cc b/libinterp/corefcn/pr-output.cc
--- a/libinterp/corefcn/pr-output.cc
+++ b/libinterp/corefcn/pr-output.cc
@@ -19,36 +19,33 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
-#include <cstdio>
-#include <cstring>
-
 #include <iomanip>
 #include <iostream>
+#include <limits>
+#include <list>
 #include <sstream>
 #include <string>
 
 #include "Array-util.h"
 #include "CMatrix.h"
 #include "Range.h"
 #include "cmd-edit.h"
 #include "dMatrix.h"
 #include "lo-mappers.h"
 #include "lo-math.h"
 #include "mach-info.h"
 #include "oct-cmplx.h"
 #include "quit.h"
-#include "str-vec.h"
 
 #include "Cell.h"
 #include "defun.h"
 #include "error.h"
 #include "errwarn.h"
 #include "ovl.h"
 #include "oct-stream.h"
 #include "octave-preserve-stream-state.h"
diff --git a/libinterp/corefcn/sysdep.cc b/libinterp/corefcn/sysdep.cc
--- a/libinterp/corefcn/sysdep.cc
+++ b/libinterp/corefcn/sysdep.cc
@@ -19,21 +19,18 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
 #include <cstddef>
-#include <cstdio>
-#include <cstdlib>
-#include <cstring>
+//#include <cstdlib>
 
 #include <iostream>
 #include <string>
 
 #if defined (HAVE_TERMIOS_H)
 #  include <termios.h>
 #elif defined (HAVE_TERMIO_H)
 #  include <termio.h>
@@ -62,17 +59,16 @@ along with Octave; see the file COPYING.
 #endif
 
 #include "cmd-edit.h"
 #include "file-ops.h"
 #include "lo-mappers.h"
 #include "lo-math.h"
 #include "mach-info.h"
 #include "oct-env.h"
-#include "quit.h"
 #include "unistd-wrappers.h"
 #include "unsetenv-wrapper.h"
 
 #include "builtin-defun-decls.h"
 #include "Cell.h"
 #include "defun.h"
 #include "display.h"
 #include "error.h"
diff --git a/libinterp/corefcn/variables.h b/libinterp/corefcn/variables.h
--- a/libinterp/corefcn/variables.h
+++ b/libinterp/corefcn/variables.h
@@ -33,24 +33,22 @@ class octave_value_list;
 class octave_builtin;
 class string_vector;
 
 namespace octave
 {
   class tree_identifier;
 }
 
-#include <cfloat>
-
 #include <limits>
 #include <string>
 
 #include "lo-ieee.h"
 
-#include "ov.h"
+//#include "ov.h"
 #include "ov-builtin.h"
 #include "symtab.h"
 
 extern OCTINTERP_API void clear_mex_functions (void);
 
 extern OCTINTERP_API octave_function *
 is_valid_function (const octave_value&, const std::string& = "",
                    bool warn = false);
diff --git a/libinterp/corefcn/xnorm.cc b/libinterp/corefcn/xnorm.cc
--- a/libinterp/corefcn/xnorm.cc
+++ b/libinterp/corefcn/xnorm.cc
@@ -21,26 +21,21 @@ along with Octave; see the file COPYING.
 */
 
 // author: Jaroslav Hajek <highegg@gmail.com>
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cassert>
-#include <cfloat>
-#include <cmath>
-
 #include "oct-norm.h"
 
-#include "error.h"
+#include "errwarn.h"
+#include "ov.h"
 #include "xnorm.h"
-#include "ov.h"
-#include "errwarn.h"
 
 octave_value xnorm (const octave_value& x, const octave_value& p)
 {
   octave_value retval;
 
   bool isvector = (x.columns () == 1 || x.rows () == 1);
   bool iscomplex = x.is_complex_type ();
   bool issparse = x.is_sparse_type ();
diff --git a/libinterp/dldfcn/__glpk__.cc b/libinterp/dldfcn/__glpk__.cc
--- a/libinterp/dldfcn/__glpk__.cc
+++ b/libinterp/dldfcn/__glpk__.cc
@@ -20,17 +20,16 @@ along with Octave; see the file COPYING.
 <http://www.gnu.org/licenses/>.
 
 */
 
 #if defined (HAVE_CONFIG_H)
 #  include "config.h"
 #endif
 
-#include <cfloat>
 #include <ctime>
 
 #include <limits>
 
 #include "Array.h"
 #include "chMatrix.h"
 #include "dColVector.h"
 #include "dMatrix.h"
